dom-fetch-client/src/main/scala/org/http4s/dom/FetchClient.scala

```diff
@@ -20,30 +20,71 @@ package dom
 import cats.effect.Async
 import cats.effect.Resource
 import cats.syntax.all._
+import cats.effect.syntax.all._
+import java.util.concurrent.TimeoutException
 import org.http4s.client.Client
+import org.http4s.headers.{`Referer`}
 import org.scalajs.dom.crypto._
+import org.scalajs.dom.experimental.AbortController
 import org.scalajs.dom.experimental.Fetch
 import org.scalajs.dom.experimental.Headers
 import org.scalajs.dom.experimental.HttpMethod
 import org.scalajs.dom.experimental.RequestInit
+import org.scalajs.dom.experimental.{Response => FetchResponse}
+import scala.concurrent.duration._
 
 object FetchClient {
 
-  def apply[F[_]](implicit F: Async[F]): Client[F] = Client[F] { req: Request[F] =>
+  private[dom] def makeClient[F[_]](
+      requestTimeout: Duration,
+      options: FetchOptions
+  )(implicit F: Async[F]): Client[F] = Client[F] { (req: Request[F]) =>
     Resource.eval(req.body.chunkAll.filter(_.nonEmpty).compile.last).flatMap { body =>
       Resource
         .makeCase {
-          F.fromPromise {
-            F.delay {
-              val init = new RequestInit {}
+          F.delay(new AbortController()).flatMap { abortController =>
+            val requestOptions = req.attributes.lookup(FetchOptions.Key)
+            val mergedOptions = requestOptions.fold(options)(options.overrideWith)
 
-              init.method = req.method.name.asInstanceOf[HttpMethod]
-              init.headers = new Headers(toDomHeaders(req.headers))
-              body.foreach { body =>
-                init.body = arrayBuffer2BufferSource(body.toJSArrayBuffer)
-              }
+            val init = new RequestInit {}
 
-              Fetch.fetch(req.uri.renderString, init)
+            init.method = req.method.name.asInstanceOf[HttpMethod]
+            // Referer header is converted to a Fetch parameter below, since it's a forbidden header.
+            // See https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name
+            init.headers = new Headers(
```

Comment:
This would be great for performance, but also makes me nervous about the one browser that does something annoying here!

Reply:
[The spec](https://fetch.spec.whatwg.org/#headers-class) seems to indicate that forbidden headers will be silently ignored if appended. Ideally every browser implements the spec 🙃 

> Otherwise, if headers’s guard is "request" and name is a forbidden header name, return.

Reply:
Hmm, it seems there is no guard when we do `new Headers`

> When a new Headers object is created using the Headers() constructor, its guard is set to none (the default).

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Basic_concepts#guard

So, seems the headers _will_ be appended, and it's up to fetch itself to not balk.